source/common/tracing/zipkin/zipkin_core_types.h

```diff
@@ -240,17 +246,22 @@ class BinaryAnnotation : public ZipkinBase {
   /**
    * Sets the annotation's endpoint attribute (copy semantics).
    */
-  void setEndpoint(const Endpoint& endpoint) { endpoint_.value(endpoint); }
+  void setEndpoint(const Endpoint& endpoint) { endpoint_ = endpoint; }
 
   /**
    * Sets the annotation's endpoint attribute (move semantics).
    */
-  void setEndpoint(const Endpoint&& endpoint) { endpoint_.value(endpoint); }
+  void setEndpoint(const Endpoint&& endpoint) { endpoint_ = endpoint; }
 
   /**
    * @return true of the endpoint attribute has been set, or false otherwise.
    */
-  bool isSetEndpoint() const { return endpoint_.valid(); }
+  bool isSetEndpoint() const {
+    if (endpoint_) {
```

Comment:
I assume you'll replace with `has_value` given the comment re: OS X flake? Thanks.

Reply:
Sure, will just have to revert the last commit.

Reply:
This one still needs fixing.